styling issues addressed across app and error handling for homepage a…

GlenMac90 opened the pull request:
Hi @cdsalty @snlaight On the pr I have fixed css issues through the app and added error handling for the homepage fetch request

Commit:
styling issues addressed across app and error handling for homepage added

GlenMac90 commented:
I have deleted the other vercel deployment

cdsalty commented on hooks/index.ts:
```diff
@@ -21,3 +27,37 @@ export async function caseStudyFullInfo() {
+
+export async function getHomePageData() {
+  try {
+    const caseStudies = await caseStudyInfoHomepage();
+    const testimonials = await getTestimonials();
+    const workExperienceList = await getWorkExperiences();
+    const services = await getServicesProvided();
+    const mySkills = await getMySkills();
```
You are doing great. There are a couple of things we want to consider here. 

1. Using Promise.all would allow these asynchronous operations to execute concurrently, meaning they all get initiated almost at the same time, and the function will wait for all to complete. This can generally lead to faster load times as well. 
An example to consider:
```
 try {
    const results = await Promise.all([
      caseStudyInfoHomepage(),
      getTestimonials(),
      getWorkExperiences(),
      getServicesProvided(),
      getMySkills(),
    ]);

    const [caseStudies, testimonials, workExperienceList, services, mySkills] = results;
    ...
    ...
```
2. Error Handling: The way it is currently set up, if one API call fails, none of the data gets returned. Consider handling errors for individual requests so that the failure of one doesn't impact the others.

3. Console.error: The error is logged to the console, but you should return something more meaningful. 
`return { error: error.message || "An unknown error occurred." };`

GlenMac90 replied:
Hi @cdsalty Thanks a lot I think I've got this working a lot better now. I'm using Promise.allSettled which is also letting me go through each operation and even if one fails the others will still fire and return something if it is able to

cdsalty commented on utils/index.ts:
```diff
@@ -18,3 +19,21 @@ export const emailFormSchema = z.object({
+export const reorderWorkProcesses = (
+  workProcesses: WorkProcessSymbolType[]
+): WorkProcessSymbolType[] => {
+  const desiredOrder = [
+    "Definition",
+    "Design",
+    "Planning",
+    "Development",
+    "Deployment",
+  ];
```
Instead of manually creating a "desired Order," another approach would first be importing the hard-coded data and then using the `sort` method to render the desired order. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

GlenMac90 replied:
I've changed this to use the sort method instead. It's working well now

cdsalty replied:
Excellent job buddy!

cdsalty reviewed (changes requested):
This looks great @GlenMac90!  I've identified a few areas for improvement that should enhance both the quality and performance of the code. Please review the recommended changes I included. Don't hesitate to reach out if you need anything or have any questions. 
Keep up the outstanding work!

cdsalty commented:
Also, @GlenMac90, remember to address the error messages in your dev console or the terminal... 

`Warning: Invalid DOM property `clip-path`. Did you mean `clipPath`?`

```
Read more: https://nextjs.org/docs/messages/next-image-upgrade-to-13
Image with src "https://cdn.sanity.io/images/v1gyd2rq/production/8934a818a16cdd644434ccfe8e3753b834f6cfec-504x972.png" has legacy prop "objectFit". Did you forget to run the codemod?
```

cdsalty commented:
Sorry, one more @GlenMac90... Remove the extra space that you have on the outside of your icons. It should be more aligned with the content being rendered below it. See the screenshot I included...

<img width="1241" alt="Screen Shot 2023-09-20 at 12 06 10 AM" src="https://github.com/GlenMac90/portfolio/assets/33855399/7f745e0d-85db-4dc2-adc1-363ad0e8c86d">

Commit:
issues addressed

GlenMac90 commented:
@cdsalty Thanks again Christopher. I have also fixed the styling as suggested above. I'm using flex-wrap now instead of grid. I think it is way more responsive and changes with the screen size much more smoothly. 

I have also changed the legacy props that I was using. I just needed to put them into a style object instead of using objectFIt as a prop.

Commit:
changed objectFit to fill

cdsalty commented on hooks/index.ts:
```diff
@@ -21,3 +27,51 @@ export async function caseStudyFullInfo() {
+  const mySkills =
+    mySkillsResult.status === "fulfilled" ? mySkillsResult.value : [];
+
+  const errors = results
+    .filter((result) => result.status === "rejected")
+    .map((result) => (result as PromiseRejectedResult).reason);
+
+  if (errors.length) {
+    console.error("Some errors occurred during data fetching:", errors);
+  }
```
Great job! I am very proud of your approach to this suggestion. Not only did you implement the changes with promise.all(), you took it a step further to implement allSettled(). Again, Great work!!! I will include the main difference between the two below. Also, I am including another approach for your error checking. You might need to adjust it but this approach should log the specific errors while running the map function first... 

- Use Promise.all when you need all promises to succeed and want to fail fast if any of the promises reject.
- Use Promise.allSettled when you want to know the outcome of each promise and don't need all of them to succeed.


```
  // Log specific errors
  const errors = results
    .map((result, index) => {
      if (result.status === "rejected") {
        const reason = (result as PromiseRejectedResult).reason;
        console.error(`Error fetching data for index ${index}:`, reason);
        return reason;
      }
      return null;
    })
    .filter((e) => e !== null);
```

cdsalty commented on app/components/LaptopMockup.tsx:
```diff
@@ -29,7 +29,7 @@ const LaptopMockup = ({ imageSrc, isHomepage = false }: MockupProps) => {
           src={imageSrc}
           alt="desktop image of project"
-          objectFit="cover"
+          style={{ objectFit: "fill" }}
```
The width and height being passed are serving as placeholders in a way... If you will be in office hours today, we can look at it and try another approach. I did reach out to Mentor Santiago and he did verify that "_Internal static images in the public folder don’t need height & width set , as next sets this on its own because it can read them at runtime (unless your eslint settings prevent you from doing this), but images from external APIs can’t be read at runtime and so next requires a height and width to explicitly be set on those._"


```
<Image
     fill
     src="/e x a m pl e . p n g "  // to prevent it from rendering here...
     sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
 />

```

cdsalty reviewed (approved):
Outstanding work @GlenMac90! I love how you took my suggestion and improved upon it by using promise.allSettled().Very proud of how well you're doing and how good your portfolio is coming along!
